holmes/plugins/toolsets/opensearch/opensearch.py

```diff
@@ -243,3 +230,12 @@ def get_example_config(self) -> Dict[str, Any]:
             ]
         )
         return example_config.model_dump()
+
+    def init_config(self, config: dict[str, Any]):
+        os_config = OpenSearchConfig(**config)
+        for cluster in os_config.opensearch_clusters:
+            logging.info("Setting up OpenSearch client")
+            cluster_kwargs = cluster.model_dump()
+            client = OpenSearchClient(**cluster_kwargs)
+            if client.client.cluster.health(params={"timeout": 5}):
+                self.clients.append(client)
```

Comment:
Putting the health check inside the config is fine by me be isn't what you want to avoid by separating the config?

What is important though is to make sure that if more than one cluster cannot be initialized that all the failures are reported to the user for the same reason that I mentioned about the Kafka clusters initialisation.

Looks like there is a bug here as well and I've recorded it separately: there is no error if a cluster reports unhealthy but it won't be part of the initialized clients.

Reply:
@mainred I think we need to be aligned with the previous behavior
I think previously, it was checking health on all clusters.
If a cluster is healthy - it's added to the list.
If not - the error is returned.

Currently, I think there are 2 issues:
1. The loop in line 240, is not with try/except - so any error will not be caught and will stop the execution
2. I think there's another loop, in line 247 - looks like almost the same code is duplicated - is this on purpose?